app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -0,0 +1,163 @@
+import mozilla.components.lib.state.ext.observeAsComposableState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+
+/**
+ * A screen for various settings related to wallpapers.
+ *
+ * @param wallpaperManager Manager for retrieving and updating current wallpaper.
+ */
+@Composable
+fun WallpaperSettings(
+    appStore: AppStore
+) {
+    val selectedWallpaper = appStore.observeAsComposableState { state -> state.wallpaper }.value
+    Surface(color = FirefoxTheme.colors.layer2) {
+        WallpaperThumbnails(
+            wallpapers = Wallpaper.values().toList(),
+            selectedWallpaper = selectedWallpaper ?: Wallpaper.NONE,
+            onSelectionChanged = { appStore.dispatch(AppAction.UpdateWallpaper(it)) }
+        )
+    }
+}
+
+/**
+ * A grid of selectable wallpaper thumbnails.
+ *
+ * @param wallpapers Wallpapers to add to grid
+ * @param selectedWallpaper The currently selected wallpaper
+ * @param numColumns The number of columns that will occupy the grid.
+ * @param onSelectionChanged Action to take when a new wallpaper is selected.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+private fun WallpaperThumbnails(
+    wallpapers: List<Wallpaper>,
+    selectedWallpaper: Wallpaper,
+    numColumns: Int = 3,
+    onSelectionChanged: (Wallpaper) -> Unit,
+) {
+    LazyVerticalGrid(
+        cells = GridCells.Fixed(numColumns),
+        modifier = Modifier.padding(vertical = 30.dp, horizontal = 20.dp)
+    ) {
+        items(wallpapers) { wallpaper ->
+            WallpaperThumbnailItem(
+                wallpaper = wallpaper,
+                isSelected = selectedWallpaper == wallpaper,
+                onSelectionChanged = onSelectionChanged
+            )
+        }
+    }
+}
+
+/**
+ * A single wallpaper thumbnail.
+ *
+ * @param wallpaper The wallpaper to display.
+ * @param isSelected Whether the wallpaper is currently selected.
+ * @param aspectRatio The ratio of height to width of the thumbnail
+ * @param onSelectionChanged Action to take when wallpaper is selected.
+ */
+@Composable
+private fun WallpaperThumbnailItem(
+    wallpaper: Wallpaper,
+    isSelected: Boolean,
+    aspectRatio: Float = 1.1f,
+    onSelectionChanged: (Wallpaper) -> Unit
+) {
+    val thumbnailShape = RoundedCornerShape(8.dp)
+    val border = if (isSelected) {
+        Modifier.border(
+            BorderStroke(width = 2.dp, color = FirefoxTheme.colors.borderAccent),
+            thumbnailShape
+        )
+    } else {
+        Modifier.border(
+            BorderStroke(width = 1.dp, color = FirefoxTheme.colors.borderDefault),
+            thumbnailShape
+        )
+    }
+    val background = if (wallpaper == Wallpaper.NONE) {
+        Modifier.background(color = FirefoxTheme.colors.layer1)
+    } else {
+        Modifier
+    }
+    Box(
```

Comment:
I would add new line before to visually separate out the variable declarations from the actual start of the compose ui.